overlord/servicestate/quota_handlers.go

```diff
@@ -595,7 +601,37 @@ func ensureSnapServicesStateForGroup(st *state.State, grp *quota.Group, opts *en
 	return restartSnapServices(st, nil, appsToRestartBySnap, nil)
 }
 
+func ensureGroupIsNotMixed(group string, allGrps map[string]*quota.Group) error {
+	grp, ok := allGrps[group]
+	if ok {
+		if len(grp.SubGroups) != 0 && len(grp.Snaps) != 0 {
+			return fmt.Errorf("quota group %q has mixed snaps and sub-groups, which is not supported anymore. please remove it and create it again to make any modifications", group)
+		}
+	}
+	return nil
+}
+
+func ensureGroupHasNoSubgroups(group string, allGrps map[string]*quota.Group) error {
+	grp, ok := allGrps[group]
+	if ok {
+		if len(grp.SubGroups) != 0 {
+			return fmt.Errorf("cannot mix snaps and sub groups in the group %q", group)
```

Comment:
This error message doesn't really match the check. I get that it's equivalent in the context that it's called (because it's called before trying to add a snap to a group) but, when just looking at this function, it looks out of place. If it's reused in a different context, the error message would be wrong.

Reply:
Really good point, i'll refactor this a bit to make more sense